include/deal.II/base/numbers.h

```diff
@@ -264,6 +265,7 @@ namespace numbers
      * Return the complex-conjugate of the given number.
      */
     static
+    DEAL_II_CUDA_HOST_DEV
```

Comment:
This is wrong. `std::complex` does not exist in cuda.

Reply:
Really? I mean that's part of C++. They may not have native support in CUDA for these kinds of data types, but is it actually wrong to put `DEAL_II_CUDA_HOST_DEV` here?

Reply:
What do you mean it is part of C++? CUDA supports C++ but it doesn't support most of what is in the standard library. You can't use `std::sort` for example because you would need to add `__host__ __device__` to the declaration of `std::sort`.  Just to be sure I tried to create a complex number on the GPU and I get the following error
``` c++
error: calling a constexpr __host__ function("std::complex<double> ::complex") from a __global__ function("norm_kernel") is not allowed
```
If we want to support complex in CUDA, we need to use CUDA's own complex type.

Reply:
I see -- I don't think I realized this. I'll remove the declaration later today.